### DISCUSSION TITLE
Add support for Mailpit as the mail trapping solution

### DISCUSSION THREAD

**Discussion by bap14** [wardenenv/warden#780] [Ideas]
Mailhog is effectively unmaintained / abanonded. While it does still work it being unmaintained increases risk of issues arising as it ages, and any security vulnerabilities unpatched. The maintainer of Mailhog has self-admitted they don't have time for it anymore (see https://github.com/mailhog/MailHog/issues/442#issuecomment-1493415258 ).

The suggestion is to switch to the successor Mailpit. It is a fork of Mailhog, kessp the same interface, but generally keeps it working and updated (last release was June 22, 2024). It still provides its own sendmail client, and can be used as an SMTP client. The port values are even the same. It's a literal drop-in replacement (similar to MariaDB for MySQL).

I've created a PR that does this [PR#779](https://github.com/wardenenv/warden/pull/779). It will add a new environment variable called `WARDEN_MAIL_SERVICE` which takes one of two values: `mailhog` or `mailpit`. By default it sticks with mailhog. A user can edit the `~/.warden/.env` file to specify the environment variable with a value of "mailpit" to start using Mailpit instead.

**Comment by bap14** [comment]
This also adjusts the hostname for the service to be `webmail.warden.test` so it's one host regardless of service used.

**Reply by navarr** [reply]
Thank you for the contribution!  There are two thoughts I have:

1. Does it really make sense to make this a flag? Or does it make more sense to replace wholesale? This feels like something the project can afford to be opinionated about without making any developers lives worse.
2. Regardless of any changes, I think it would be beneficial for `mailhog.warden.test` to continue working.  This should be accomplishable by `Host(a) || Host(b)` as the container tag.

**Reply by sprankhub** [reply]
1. I vote for replacing it completely. We have enough configuration already :D 
2. IMHO, `webmail.warden.test` should be the new default and `mailhog.warden.test` should redirect to it for backwards compatibility.

**Reply by navarr** [reply]
> 2. IMHO, `webmail.warden.test` should be the new default and `mailhog.warden.test` should redirect to it for backwards compatibility.

That's a good thought and I agree completely.  It will require a second traefik router and service added to Mailpit though; but I suppose thats fine.

**Reply by bap14** [reply]
When I have time (should be some time soon), I'll try and incorporate this feedback. Thanks!

**Reply by navarr** [reply]
@bap14 Appreciate ya!

**Reply by bap14** [reply]
@navarr @sprankhub Updates made. Mailpit now replaces Mailhog, and going to mailhog.warden.test 301 redirects to webmail.warden.test. Not sure why I need to format the yaml labels list using the object notation, but it wouldn't redirect otherwise.
